lms/djangoapps/edraak_certificates/views.py

```diff
@@ -1,59 +1,93 @@
 import logging
-import os
 
-from wand.image import Image
+import os
+from dateutil.relativedelta import relativedelta
 
-from django.http import HttpResponse
+from django.http import HttpResponse, JsonResponse
 from django.shortcuts import redirect
 from django.conf import settings
 from django.core.urlresolvers import reverse
 from django.core.servers.basehttp import FileWrapper
-from django.core.files.temp import NamedTemporaryFile
 from django.contrib.auth.decorators import login_required
 from django.db import transaction
 
-from dateutil.relativedelta import relativedelta
 from edxmako.shortcuts import render_to_response
 from xmodule.modulestore.django import modulestore
 
 from opaque_keys.edx import locator
 from datetime import date
+
+from certificates.api import \
+    certificate_downloadable_status, generate_user_certificates
+from certificates.models import CertificateStatuses, \
+    GeneratedCertificate, certificate_status_for_student
+from certificates.views import render_cert_by_uuid
 from .utils import generate_certificate
 from edraak_misc.utils import is_student_pass, is_certificate_allowed
 
-
 logger = logging.getLogger(__name__)
 
-@login_required
-def issue(request, course_id):
-    return render_to_response('edraak_certificates/issue.html', {
-        'course_id': course_id,
-    })
-
 
 @transaction.non_atomic_requests
 @login_required
-def view(request, course_id):
-    user = request.user
+def issue(request, course_id):
+    student = request.user
 
+    # Extracts CourseLocator Object from course ID.
     course_key = locator.CourseLocator.from_string(course_id)
+    # Extract the CourseOverview object of the course.
     course = modulestore().get_course(course_key)
 
-    if is_certificate_allowed(user, course) and is_student_pass(user, request, course_id):
-        template = 'edraak_certificates/view.html'
-    else:
-        template = 'edraak_certificates/fail.html'
+    # Check the status of GeneratedCertificate in XQueue.
+    certificate_status = \
+        certificate_status_for_student(student, course.id)['status']
+
+    template = None
```

Comment:
This cannot be null, so you might consider using `edraak_certificates/fail.html` instead. It'll cause a 500 if your view failed to detect the template.